harvester_e2e_tests/integration/test_hosts.py

```diff
@@ -67,3 +178,159 @@ def test_host_poweron_state(api_client, host_state, wait_timeout):
+            from {src_host} to another node"
+    )
+    assert 'Running' == data['status']['phase'], (
+        f"Failed to migrate VM {focal_vm['namespace']}/{focal_vm['name']}, \
+            it's not running after migration"
+    )
+
+    # teardown
+    code, data = api_client.hosts.maintenance_mode(src_host, enable=False)
+    assert 200 == code, (
+        f"Failed to disable maintenance mode on node {src_host} with error: {code}, {data}",
+    )
+
+
+@ pytest.mark.hosts
+@ pytest.mark.p0
+@ pytest.mark.dependency(depends=["host_poweroff, host_poweron"])
+def test_poweroff_node_trigger_vm_migrate(api_client, host_state, focal_vm, wait_timeout,
+                                          available_node_names, vm_force_reset_policy):
+    assert 2 <= len(available_node_names), (
+        f"The cluster only have {len(available_node_names)} available node. \
+            It's not enough for migration test."
+    )
+
+    code, data = api_client.vms.get_status(focal_vm['name'], focal_vm['namespace'])
+    assert 200 == code, (
+        f"Can't get VMI {focal_vm['namespace']}/{focal_vm['name']} with error: {code}, {data}"
+    )
+    src_host = data['status']['nodeName']
+
+    # poweroff host
+    _, node = api_client.hosts.get(src_host)
+    node_ip = next(val["address"] for val in node['status']['addresses']
+                   if val["type"] == "InternalIP")
+    rc, out, err = host_state.power(node['id'], node_ip, on=False)
+    assert rc == 0, (f"Failed to PowerOff node {node['id']} with error({rc}):\n"
+                     f"stdout: {out}\n\nstderr: {err}")
+    sleep(host_state.delay)  # Wait for the node to disappear
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        _, metric = api_client.hosts.get_metrics(node['id'])
+        if metric.get("status") == 404:
+            break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"Node {node['id']} still available after PowerOff script executed"
+            f", script path: {host_state.path}"
+        )
+
+    # check vm is migrated to another host
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        code, data = api_client.vms.get_status(focal_vm['name'], focal_vm['namespace'])
+        if data.get('status', {}).get('phase', "") == "Running":
+            assert src_host != data['status']['nodeName'], (
+                f"Failed to migrate vm {focal_vm['namespace']}/{focal_vm['name']} \
+                    from {src_host} to another node"
+            )
+            break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"VM {focal_vm['namespace']}/{focal_vm['name']} can't be Running \
+                with {wait_timeout} timed out\n"
+            f"Got error: {code}, {data}"
+        )
+
+    # teardown
+    rc, out, err = host_state.power(node['id'], node_ip, on=True)
```

Comment:
Same concern, if the test fails, the host will remain power off.

Reply:
same as above, we already check that cluster will have at least 2 nodes, so even 1 node is powered off, following test cases (not requires multiple nodes) will still able to be tested.

Reply:
I'm not only concerned about the next subsequent test but other tests in the entire test suite which will run after this test case. We don't want to have a powered off node when other tests are running assuming that all the nodes are up.

Do we have any fixture or are we resetting the env before starting the other module test cases?

Reply:
Yeah we can make a module scoped fixture to do it, but if we do that, which means the failed environment is not kept.

there were a issue that VM will not migrate to another host automatically when the node powered off.  But the VM will try migrating again after power on the node (and the migration will success).

for other test suits, we should always check them if nodes’ status is the prerequisite of testing

Reply:
I also have concerns about the *vm-force-reset-policy*, I think might need to discuss the test case further to make sure steps and criteria is suitable (and we should update manual tests too)